R/clean.R

```diff
@@ -0,0 +1,82 @@
+#' Clean Rust binaries and package cache.
+#'
+#' Removes Rust binaries (such as `.dll`/`.so` libraries), C wrapper object files,
+#' invokes `cargo clean` to reset cargo target directory
+#' (found by default at `pkg_root/src/rust/target/`).
+#' Useful when Rust code should be recompiled from scratch.
+#' @param path \[ string \] Path to the package root.
+#' @export
+clean <- function(path = ".") {
+  root <- rprojroot::find_package_root_file(path = path)
+
+  rust_folder <- normalizePath(
+    file.path(root, "src", "rust"),
+    winslash = "/",
+    mustWork = FALSE
+  )
+
+  toml_path <- normalizePath(
+    file.path(rust_folder, "Cargo.toml"),
+    winslash = "/",
+    mustWork = FALSE
+  )
+
+  # Note: This should be adjusted if `TARGET_DIR` changes in `Makevars`
+  target_dir <- normalizePath(
+    file.path(rust_folder, "target"),
+    winslash = "/",
+    mustWork = FALSE
+  )
+
+  if (!file.exists(toml_path)) {
+    ui_throw(
+      "Unable to clean binaries.",
+      c(
+        bullet_w(
+          "{.file Cargo.toml} not found in {.path {rust_folder}}."
+        )
+      )
+    )
+  }
+
+  env <- rlang::current_env()
+  cargo_envvars <- get_cargo_envvars()
+
+  args <- c(
+    "clean",
+    glue("--manifest-path={toml_path}"),
+    glue("--target-dir={target_dir}"),
+    if (tty_has_colors()) {
+      "--color=always"
+    } else {
+      "--color=never"
+    },
+    "--quiet"
+  )
```

Comment:
I still think we should do this..

Reply:
Alright, I've tested things. I see Rust Analyzer use `-p` with their invocation of `cargo` commands , but I think this here suffices. The problem occurs when you override the target directory, and you have one directory with all build artefacts. This speeds up compilation of things, and mitigates the growth of cargo caches. But what it does, is it causes an issue because plain `cargo clean` becomes ambiguous, and it starts to clean-up *everything*.
So here's a screenshot: 

![image](https://user-images.githubusercontent.com/1063624/168617835-da8c49cd-f5cc-4ec1-b4b1-eb93532c1161.png)

When I run this, it takes it **a minute**, and it even fails because I'm currently using a `nu`-shell, that is built on `cargo` and it cannot delete it, because I'm actively using it (only Windows folks knows of this lunacy). 

However, when you also specify the target directory, then your implementation works just fine. And perhaps it suffices just to specify the two things, the `Cargo.toml` and the target-directory.

Reply:
I tested it, and you do not need to pass `--package`. I tested your command, and it works, because you pass the two args. But I believe that we should pass `--package`, I just don't have a concrete reason for it.

Reply:
Are you ok if we leave it as is for now?